consulate/api/kv.py

```diff
@@ -379,11 +393,17 @@ def _set_item(self, item, value, flags=None, replace=True):
         if index is None:
             return True
 
-        query_params = {'cas': index}
+        if params and isinstance(params, dict):
+            query_params = params
+        else:
+            query_params = {'cas': index}
+
         if flags is not None:
             query_params['flags'] = flags
+
         response = self._adapter.put(self._build_uri([item], query_params),
                                      value)
         if not response.status_code == 200 or not response.body:
-            raise KeyError(
```

Comment:
How does this not break the contract with regard to a KeyError expectation if the response could not be sent?